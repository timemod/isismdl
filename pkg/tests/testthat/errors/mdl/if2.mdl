? if statements with type mismatch (logical and reals mixed).
? error should be detected by readmdl

ident x1 = if (x) then 2 else 2;
ident x2 = if (3 .or. 2) then 2 else 2;
ident x3 = if (3 = 2) then 2 else 2 = 4;
ident x4 = if (.not. 2) then 2 else 3;
ident x5 = if (x = 2) then 2 else (3 + (2 = 1));

? the next equation does not result in an error:
? the logical value is convert to a number
ident x6 = 3 = 2; 
