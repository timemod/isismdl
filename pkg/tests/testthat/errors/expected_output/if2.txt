Isis Model Compiler 3.00
Compiling model ...
       6 equations processed
Ordering equations ...
Writing MIF file ...
Writing cross-reference file ...
End compilation
Reading mif file ...
Model with     6 equations read
Checking Model-code ...
Logical operand expected in equation x1
Logical operand expected in equation x2
Type mismatch between Then and Else expression in equation x3
Logical operand expected in equation x4
Arithmetic operand expected in equation x5
Errors encountered in model ...
